Common/cpp/NativeModules/NativeReanimatedModule.cpp

```diff
@@ -161,6 +162,21 @@ void NativeReanimatedModule::scheduleOnUI(
   });
 }
 
+jsi::Value NativeReanimatedModule::executeOnUIRuntimeSync(
+    jsi::Runtime &rt,
+    const jsi::Value &worklet) {
+  auto shareableWorklet = extractShareableOrThrow<ShareableWorklet>(
+      rt,
+      worklet,
+      "[Reanimated] Only worklets can be executed synchronously on UI runtime.");
+  auto lock = uiWorkletRuntime_->lock();
+  jsi::Runtime &uiRuntime = uiWorkletRuntime_->getJSIRuntime();
+  auto result = uiWorkletRuntime_->runGuarded(shareableWorklet);
+  auto shareableResult = extractShareableOrThrow(uiRuntime, result);
+  lock.unlock();
+  return shareableResult->getJSValue(rt);
```

Comment:
Does it make sense to actually move this implementation to a new method in`WorkletRuntime` class and call it here?